### ISSUE TITLE
schedule s3fs (Access Denied)

### ISSUE THREAD

**Issue by phdnzdeveloper** [dask/s3fs#249]
Hi Team,

I currently use s3fs (for S3 Storage instead of Local storage) to schedule data pipelines everyday . **For first time, after initialize a s3fs.S3FileSystem() instance**, all pipelines **successfully** ran, **but next days**, **they are all failed** (_the old files still successfully run_). I found that all functions (ls(), info(),...) have **refresh** parameters. However, I still get **Access Denied** for new files, it seems consistent. Could you do any things to fix it?

Regards
Giang

**Comment by martindurant** [comment]
What version are you running? This may be fixed in fsspec and s3fs master.

**Comment by martindurant** [comment]
Also, what method do you use to auth, any special arguments to S3FileSystem? If you are using roles and STS, you might find this interesting reading: https://dev.to/li_chastina/auto-refresh-aws-tokens-using-iam-role-and-boto3-2cjf

**Comment by phdnzdeveloper** [comment]
Hi Martin,

My version is 0.3.4, I used default initialization with no parameters.
`fs= s3fs.S3FileSystem()`

**Comment by martindurant** [comment]
Sorry, I got a little lost there - could you post the code, please?

**Comment by phdnzdeveloper** [comment]
Hi Martin,

I used functions ls and open (sometimes exists) **many times in one process** in day. So I cannot track back. But I founded a solution it is clear_instance_cache(), it solved problem. However, for now, a documentation hasn't updated yet.
![image](https://user-images.githubusercontent.com/56419140/67712840-9d3af280-fa29-11e9-83c1-6c248545f2d5.png)

Additionally, an error is like this
![image](https://user-images.githubusercontent.com/56419140/67714503-e5a7df80-fa2c-11e9-8eea-1b867c8f52a0.png)
![image](https://user-images.githubusercontent.com/56419140/67714571-0a9c5280-fa2d-11e9-8825-57ca7ec93e01.png)
